cmd/root.go

```diff
@@ -250,7 +251,7 @@ func initConfig() {
 	viper.AutomaticEnv() // read in environment variables that match
 	// If a config file is found, read it in.
 	if err := viper.ReadInConfig(); err != nil {
-		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+		if !errors.As(err, &viper.ConfigFileNotFoundError{}) {
```

Comment:
`errors.Is` wouldn't match the intention here. `errors.As` verifies that we can cast the error (or any of its unwrapped errors) to a given type which works in this situation. If we used `errors.Is` we'd get false even if the type was matching because of two compared errors not being equal in terms of their `==` equality: 

```go
func TestError(t *testing.T) {
	err := viper.ReadInConfig()
	require.ErrorContains(t, err, "Config File \"config\" Not Found")

	withErrorsAs := errors.As(err, &viper.ConfigFileNotFoundError{})
	withErrorsIs := errors.Is(err, viper.ConfigFileNotFoundError{})

	assert.True(t, withErrorsAs)
	assert.False(t, withErrorsIs)
}
```

Reply:
`errors.Is` could work if `viper.ConfigFileNotFoundError` implemented `Is(error) bool` interface and just used `errors.As` in it, ignoring the values stored in the error, caring only about the type.

Reply:
```go
type ConfigFileNotFoundError struct {
	filePath string
}

func (c ConfigFileNotFoundError) Error() string {
	return fmt.Sprintf("config file %q not found", c.filePath)
}

func (c ConfigFileNotFoundError) Is(target error) bool {
	_, ok := target.(ConfigFileNotFoundError)
	return ok
}

func TestErrorIs(t *testing.T) {
	err := fmt.Errorf("error occurred: %w", ConfigFileNotFoundError{filePath: "config"})
	withErrorsIs := errors.Is(err, ConfigFileNotFoundError{})
	require.True(t, withErrorsIs)
}
```
The above fails if `func (c ConfigFileNotFoundError) Is(target error) bool` is commented out.

Reply:
I see ok, thanks for the example!